uaclient/system.py

```diff
@@ -17,6 +17,123 @@
 # N.B. this relies on the version normalisation we perform in get_platform_info
 REGEX_OS_RELEASE_VERSION = r"(?P<release>\d+\.\d+) (LTS )?\((?P<series>\w+).*"
 
+RE_KERNEL_UNAME = (
+    r"^"
+    r"(?P<version>[\d]+[.-][\d]+[.-][\d]+)"
+    r"-"
+    r"(?P<abi>[\d]+)"
+    r"-"
+    r"(?P<flavor>[A-Za-z0-9_-]+)"
+    r"$"
+)
+RE_KERNEL_PROC_VERSION_SIGNATURE = (
+    r"^"
+    r"(?P<version>[\d]+[.-][\d]+[.-][\d]+)"
+    r"-"
+    r"(?P<abi>[\d]+)"
+    r"[.-]"
+    r"(?P<subrev>[\d]+)"
+    r"(~(?P<hwerev>[\d.]+))?"
+    r"-"
+    r"(?P<flavor>[A-Za-z0-9_-]+)"
+    r"$"
+)
+RE_KERNEL_VERSION_SPLIT = (
+    r"^"
+    r"(?P<major>[\d]+)"
+    r"[.-]"
+    r"(?P<minor>[\d]+)"
+    r"[.-]"
+    r"(?P<patch>[\d]+)"
+    r"$"
+)
+
+KernelInfo = NamedTuple(
+    "KernelInfo",
+    [
+        ("uname_release", str),
+        ("proc_version_signature_full", str),
+        ("proc_version_signature_version", str),
+        ("version", str),
+        ("major", str),
+        ("minor", str),
+        ("patch", str),
+        ("abi", str),
+        ("subrev", str),
+        ("hwerev", str),
+        ("flavor", str),
+    ],
+)
+
+
+@lru_cache(maxsize=None)
+def get_kernel_info() -> KernelInfo:
+    uname_release = os.uname().release
+
+    proc_version_signature_full = ""
+    proc_version_signature_version = ""
+    try:
+        proc_version_signature_full = load_file("/proc/version_signature")
+        proc_version_signature_version = proc_version_signature_full.split(
+            " "
```

Comment:
I think this is default parameter for split, we can omit it if we want

Reply:
Looks like with no arg it splits on "runs of consecutive whitespace" but that works too and maybe is even better if the format of that file slightly changes ever. I'll change it